### ISSUE TITLE
src directory

### ISSUE THREAD

**Issue by umutacar** [quantum-compiler/quartz#6]
I would recommend moving the code under a "src" directory
The top level would have things like 
1) virtual env's needed for python
2) documentation
3) infrastructure related scripts (e.g., for cloud setup etc if any)

Without a separate an "src" directory, it is difficult to do a number of things, for example, visually separating varioun configuration/infra matters from actual code.  It is also difficult to  search the source code because for example the virtual env code will get mixed in, which makes search (e.g., `grep -r`) effectively unusable (BAD).

**Comment by xumingkuan** [comment]
I think moving the code under a "src" directory is already done by @zikun-li in #2. Actually, all .cpp code outside this directory are deprecated after this PR and can be safely deleted.

However, the Python code is not refactored in the PR. I think one way to do the refactoring is to put them into `src/python`. WDYT?

**Comment by umutacar** [comment]
We are looking over the code with Mingkuan now. Couple of observations
-  There is code duplication, e.g., context/context.h and includes/context.h
We probably want just one of these
- We still do not have all the code under src.  we should move 
- We seem to be using "tests" to mean benchmarks and a few are for correctness
We should use "test" to mean correctess and "benchmark/experiment" to mean run-time measurement etc.
- We can set up a separate "test suite" that checks that the compiler is correct.  At each pull request, we should set up github to run these tests.  To do this, we create a .github/workflows directory and spec the tests to run

**Comment by zikun-li** [comment]
Got it. For 1 and 2, I'll delete the deprecated files, and organize all the .h and .cpp files into the `src/` directory. For 3, 4 and 5, we plan to separate all the tests files into two directories, `src/test` for correctness and `src/benchmark` for benchmark. I'll dicuss with Mingkuan to figure out how to construct test suite and benchmarks.

**Comment by xumingkuan** [comment]
Except for `src/test` and `src/benchmark` (or `src/experiment`), maybe we also want to put `python` into the `src/` directory?

**Comment by zikun-li** [comment]
That's right, I gonna move these files.

**Comment by zikun-li** [comment]
I see there is a `requirement.txt` file under the `python` directory. There should be no `python` directory after moving all python files to `src/python/`, maybe we should put the `requirement.txt` under the top directory?

**Comment by xumingkuan** [comment]
> I see there is a `requirement.txt` file under the `python` directory. There should be no `python` directory after moving all python files to `src/python/`, maybe we should put the `requirement.txt` under the top directory?

Oh maybe we should not move that. `src/` directory is for the source code, and `requirement.txt` is for configuration. Putting it under `python/` shows that it's for Python configuration.

**Comment by zikun-li** [comment]
@xumingkuan there is a `verify_equivalences.py` file under `python/` directory, is it a test file?

**Comment by xumingkuan** [comment]
No, verify_equivalences.py should be count as source code and it actually provides an API for C++ code to invoke as `system("verify_equivalences.py ...")`.

**Comment by zikun-li** [comment]
Then I'll put it into `src/python/verifier/`

**Comment by xumingkuan** [comment]
Ok sounds good.
